src/Compilers/CSharp/Portable/Syntax/SyntaxKindFacts.cs

```diff
@@ -1779,5 +1779,14 @@ public static bool IsDocumentationCommentTrivia(SyntaxKind kind)
             return kind == SyntaxKind.SingleLineDocumentationCommentTrivia ||
                 kind == SyntaxKind.MultiLineDocumentationCommentTrivia;
         }
+
+        public static bool IsIdentifierContainerDirectiveTrivia(SyntaxKind kind)
+        {
+            return kind
+                is SyntaxKind.IfDirectiveTrivia
+                or SyntaxKind.ElifDirectiveTrivia
+                or SyntaxKind.DefineDirectiveTrivia
+                or SyntaxKind.UndefDirectiveTrivia;
+        }
```

Comment:
i don't undrstand what this function is needed for, or why it was added to the compiler